MdeModulePkg/Bus/Pci/NvmExpressDxe/NvmExpress.c

```diff
@@ -1194,7 +1297,7 @@ NvmExpressDriverBindingStart (
 
   if ((Private != NULL) && (Private->Buffer != NULL)) {
     // MU_CHANGE - Support alternative hardware queue sizes in NVME driver
-    PciIo->FreeBuffer (PciIo, QueuePageCount, Private->Buffer);
+    PciIo->FreeBuffer (PciIo, AdminQueuePageCount, Private->Buffer); // MU_CHANGE - Allocate IO Queue Buffer

```

Comment:
Unchecked status here that will eventually be picked up by codeql.

```suggestion
    Status = PciIo->FreeBuffer (PciIo, AdminQueuePageCount, Private->Buffer); // MU_CHANGE - Allocate IO Queue Buffer
    if( EFI_ERROR(Status)) {
      DEBUG(DEBUG_ERROR, "%a() - Failed to Free AdminQueuePageCount %r\n", __func__, Status));
    }

```